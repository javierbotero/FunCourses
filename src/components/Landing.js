import React from 'react';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { OPINIONS } from '../constants/constants';

const Landing = () => {
  const displayOpinions = opinions => opinions.map(o => (
    <div key={o.avatar} className="opinion">
      <div style={{
        backgroundImage: o.avatar,
      }}
      />
      <div>{o.body}</div>
    </div>
  ));

  return (
    <div className="info-landing">
      <header>
        <div className="logo" />
        <div>Fun Courses</div>
      </header>
      <main>
        <section>
          <h4>Find a course of any theme</h4>
          <p>
            The best courses for you about any
            theme you can think of, taught by
            professionals.
          </p>
          <Link to="/signin">Start Free</Link>
        </section>
        <section>
          <h4>Interesting courses from reliable teachers</h4>
          <div>
            <div>
              <h4>
                All stats at your fingerprints
              </h4>
              <p>
                All the information about the number
                of students, your courses, comments
                and interactions will be available to
                you to measure your performance.
              </p>
            </div>
            <div className="stats" />
          </div>
        </section>
        <section>
          <div className="stats" />
          <div>
            <div>
              <h4>
                Talk with professionals
              </h4>
              <p>
                The best professionals around the world will
                be reachable through live courses. take the
                opportunity to enhance your skills in any matter.
              </p>
            </div>
          </div>
        </section>
        <section>
          <div>PEOPLE ARE ALREADY TAKING FUN COURSES</div>
          <h4>Look what these are saying</h4>
          <div className="opinions">
            {displayOpinions(OPINIONS)}
          </div>
        </section>
      </main>
      <footer>
        <div>
          <h4>Learn everything at home</h4>
          <Link to="/signin">Start free</Link>
        </div>
        <div>
          <div className="logo" />
          <div>Fun Courses</div>
        </div>
        <div>
          <FontAwesomeIcon icon={['fab', 'linkedin']} />
          <FontAwesomeIcon icon={['fab', 'facebook']} />
          <FontAwesomeIcon icon={['fab', 'twitter-square']} />
          <FontAwesomeIcon icon={['fab', 'instagram-square']} />
        </div>
      </footer>
    </div>
  );
};

export default Landing;
