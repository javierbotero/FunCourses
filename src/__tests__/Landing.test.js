import React from 'react';
import {
  render,
  screen,
} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Landing from '../components/Landing';

describe('Tests for Landing', () => {
  it('prints Find a course of any theme', () => {
    render(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>,
    );
    expect(screen.getByText('Find a course of any theme')).toBeInTheDocument();
  });
  it('prints Interesting courses from reliable teachers', () => {
    render(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>,
    );
    expect(screen.getByText('Interesting courses from reliable teachers')).toBeInTheDocument();
  });
  it('Prints correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Landing />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
