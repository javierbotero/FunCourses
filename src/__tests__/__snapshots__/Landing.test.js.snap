// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for Landing Prints correctly 1`] = `
<div
  className="info-landing"
>
  <header>
    <div
      className="logo"
    />
    <div>
      Fun Courses
    </div>
  </header>
  <main>
    <section>
      <h4>
        Find a course of any theme
      </h4>
      <p>
        The best courses for you about any theme you can think of, taught by professionals.
      </p>
      <a
        href="/signin"
        onClick={[Function]}
      >
        Start Free
      </a>
    </section>
    <section>
      <h4>
        Interesting courses from reliable teachers
      </h4>
      <div>
        <div>
          <h4>
            All stats at your fingerprints
          </h4>
          <p>
            All the information about the number of students, your courses, comments and interactions will be available to you to measure your performance.
          </p>
        </div>
        <div
          className="stats"
        />
      </div>
    </section>
    <section>
      <div
        className="stats"
      />
      <div>
        <div>
          <h4>
            Talk with professionals
          </h4>
          <p>
            The best professionals around the world will be reachable through live courses. take the opportunity to enhance your skills in any matter.
          </p>
        </div>
      </div>
    </section>
    <section>
      <div>
        PEOPLE ARE ALREADY TAKING FUN COURSES
      </div>
      <h4>
        Look what these are saying
      </h4>
      <div
        className="opinions"
      >
        <div
          className="opinion"
        >
          <div
            style={
              Object {
                "backgroundImage": "user1.jpg",
              }
            }
          />
          <div>
            The best place to to teach about fashion in a fun way
          </div>
        </div>
        <div
          className="opinion"
        >
          <div
            style={
              Object {
                "backgroundImage": "user2.jpeg",
              }
            }
          />
          <div>
            The best place to to teach about fashion in a fun way
          </div>
        </div>
      </div>
    </section>
  </main>
  <footer>
    <div>
      <h4>
        Learn everything at home
      </h4>
      <a
        href="/signin"
        onClick={[Function]}
      >
        Start free
      </a>
    </div>
    <div>
      <div
        className="logo"
      />
      <div>
        Fun Courses
      </div>
    </div>
    <div>
      <i
        className="fa fab,linkedin"
      />
      <i
        className="fa fab,facebook"
      />
      <i
        className="fa fab,twitter-square"
      />
      <i
        className="fa fab,instagram-square"
      />
    </div>
  </footer>
</div>
`;
